tests/async/test_network.py

```diff
@@ -230,18 +224,81 @@ def handle_something(request):
     request = await request_info.value
     assert text == "done"
     server_headers = await server_request_headers_future
-    if is_webkit and is_win:
-        # Curl does not show accept-encoding and accept-language
-        server_headers.pop("accept-encoding")
-        server_headers.pop("accept-language")
-    assert request.headers == server_headers
+    assert await request.all_headers() == server_headers
+
+
+async def test_should_report_request_headers_array(page: Page, server: Server) -> None:
+    expected_headers = []
 
+    def handle(request):
+        for key, values in request.requestHeaders.getAllRawHeaders():
+            for value in values:
+                expected_headers.append([key.decode(), value.decode()])
+        request.finish()
+
+    server.set_route("/headers", handle)
+    await page.goto(server.EMPTY_PAGE)
+    async with page.expect_request("*/**") as request_info:
+        await page.evaluate(
+            """() => fetch('/headers', {
+            headers: [
+                ['header-a', 'value-a'],
+                ['header-b', 'value-b'],
+                ['header-a', 'value-a-1'],
+                ['header-a', 'value-a-2'],
+            ]
+            })
+        """
+        )
+    request = await request_info.value
+    assert (await request.headers_array()).sort() == expected_headers.sort()
```

Comment:
```suggestion
    assert sorted(await request.headers_array()) == sorted(expected_headers)
```
This is wrong as `[].sort()` returns `None`